### DISCUSSION TITLE
How do I enable metrics?

### DISCUSSION THREAD

**Discussion by azeng528** [apache/pulsar#23733] [Q&A]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

pulsar version：3.0.2
|># java  -version
java version "17.0.9" 2023-10-17 LTS
|># cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"

### Minimal reproduce step

cluster is ok，However, the topic metrics extracted by the metrics interface are missing，After I restart the brokers and bookies of the entire cluster later, metrics will include the missing topic metrics，for example，before restart，not have the Topic of screenshot
![image](https://github.com/user-attachments/assets/153dfd73-ae3c-4ba6-955e-ae97dffd5963)

### What did you expect to see?

![image](https://github.com/user-attachments/assets/10704233-a49d-4862-9a26-30feed22c85c)
i hope the topic of screenshot is always get metrics，not  restart broker and bookie ，then get this metrics

### What did you see instead?

no

### Anything else?

1

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by azeng528** [comment]
lose metrics name: pulsar subscribtion backlog

**Comment by Shawyeok** [comment]
Could you please provide more context here, such as the `broker.conf` configuration? Have you tried reproducing this issue with the latest 3.0.x version, specifically `3.0.8`?

**Reply by azeng528** [reply]
i don't know which version have repair this bug，so i am not upgrate cluster to 3.0.8
broker.conf
metadataStoreUrl=zk:1.1.1.1:2181/cluster01
metadataStoreConfigPath=zk:1.1.1.1:2181/cluster01
configurationMetadataStoreUrl=zk:1.1.1.1:2181/cluster01
brokerServicePort=6650
webServicePort=8080
bindAddress=0.0.0.0
advertisedAddress=1.1.1.1
haProxyProtocolEnabled=false
numOrderedExecutorThreads=8
numCacheExecutorThreadPoolSize=10
enableBusyWait=false
clusterName=pulsar-cluster-01
maxTenants=0
bookkeeperMetadataServiceUri=zk+hierarchical://1.1.1.1:2181/cluster01/ledgers
topicLevelPoliciesEnabled=false

bookeeper.conf
bookiePort=3181
journalDirectories=/data1/bookie/journal1,/data1/bookie/journal2
minUsableSizeForIndexFileCreation=1073741824
advertisedAddress=1.1.1.1
allowLoopback=false
bookieDeathWatchInterval=1000
flushInterval=60000
useHostNameAsBookieID=false
bookieId=1.1.1.1:3181
metadataServiceUri=zk://1.1.1.1:2181/cluster01/ledgers

**Comment by lhotari** [comment]
Is this a cluster deployed with Apache Pulsar Helm chart? There's a known issue https://github.com/apache/pulsar-helm-chart/issues/468 . Most recent comment provides some advice: https://github.com/apache/pulsar-helm-chart/issues/468#issuecomment-2564322780 . Contributions are welcome to address the issue!
